zuul-core/src/main/java/com/netflix/zuul/netty/connectionpool/PerServerConnectionPool.java

```diff
@@ -281,6 +273,27 @@ protected void tryMakingNewConnection(
         }
     }
 
+    protected boolean checkConnectionLimit(Promise<PooledConnection> promise) {
```

Comment:
Maybe `s/checkConnectionLimit/withinConnectionLimit` ? Reads a bit nicer at the callsite

```
if (!withinConnectionLimit(promise)) {
...
}
```

or event `isWithinConnectionLimit` to mimic the waterline func below